adapters/repos/db/migrator.go

```diff
@@ -355,9 +343,10 @@ func (m *Migrator) UpdateTenants(ctx context.Context, class *models.Class, updat
 			ctx, cancel := context.WithTimeout(context.Background(), 1*time.Hour)
 			defer cancel()
 
-			if err := asLL.Load(ctx); err != nil {
+			if err := idx.loadLocalShard(ctx, name); err != nil {
```

Comment:
So interestingly doing that call and forcing to go via the lock to lookup the shard (instead of relying on the mutex inside the shard) fixes the race. Good catch.